### DISCUSSION TITLE
Problem migration with New Storage option --storage=jpa

### DISCUSSION THREAD

**Discussion by pablodiazv** [keycloak/keycloak#14890] [Archive]
I tried to migrate from postgres:11 to postgres:14 and use the new storage option --storage=jpa.
First, I did a backup from database and restore it in postgres 14 instance.

Then I start keycloak instance with: 
`start --cache-config-file=my-cache-ispn.xml --hostname-strict=false --hostname-strict-https=false --proxy edge --spi-connections-jpa-legacy-migration-strategy=update --spi-connections-jpa-legacy-initialize-empty=false --spi-connections-jpa-quarkus-migration-strategy=update --spi-connections-jpa-quarkus-initialize-empty=false`

Instead the migration, the database is initialize from zero, as if the first execution was.
Without storage option data is ok in database.

**ENVIROMENT**:
- Docker
- Keycloak version: 19.0.2
- Postgres version: 14

**Comment by tkconsulting** [comment]
See https://keycloak.discourse.group/t/upgrading-18-0-2-to-19-0-2/17883 you might be missing KC_DB property.

**Reply by pablodiazv** [reply]
I update 

> I tried to migrate from postgres:11 to postgres:14 and use the new storage option --storage=jpa. First, I did a backup from database and restore it in postgres 14 instance.
> 
> Then I start keycloak instance with: `start --cache-config-file=my-cache-ispn.xml --hostname-strict=false --hostname-strict-https=false --proxy edge --spi-connections-jpa-legacy-migration-strategy=update --spi-connections-jpa-legacy-initialize-empty=false --spi-connections-jpa-quarkus-migration-strategy=update --spi-connections-jpa-quarkus-initialize-empty=false`
> 
> Instead the migration, the database is initialize from zero, as if the first execution was. Without storage option data is ok in database.
> 
> **ENVIROMENT**:
> 
> * Docker
> * Keycloak version: 19.0.2
> * Postgres version: 14

UPDATED
I forget posted my .env files.

- **.env-postgres-keycloak**
POSTGRES_DB=keycloak
POSTGRES_USER=keycloak
POSTGRES_PASSWORD=<pass>

- **.env-postgres-keycloak**
KC_DB=postgres
KC_DB_URL_HOST=dbkeycloak
DB_DATABASE=keycloak
KC_DB_USERNAME=keycloak
KC_DB_SCHEMA=public
KC_DB_PASSWORD=<pass>
KEYCLOAK_ADMIN=<admin-user>
KEYCLOAK_ADMIN_PASSWORD=<admin-pass>
KC_HTTP_ENABLED=true
